### DISCUSSION TITLE
Provide a Zodios integration ?

### DISCUSSION THREAD

**Discussion by cyrilchapon** [pmndrs/jotai#2279] [Ideas]
In the superb [Zodios](https://www.zodios.org/) library, their [react hooks](https://www.zodios.org/docs/client/react) are powered by react-query.

I personally adore both of Zodios and Jotai worlds; and would completely fall in love if I could use both at the same time ! 😍

Opened the discussion there too => https://github.com/ecyrbe/zodios/issues/568

**Comment by dai-shi** [comment]
They have vanilla JS api: https://www.zodios.org/docs/client

It will be somewhat close to https://github.com/jotaijs/jotai-trpc
Would you or anyone be interested in maintaining jotai-zodios?

**Reply by cyrilchapon** [reply]
Hi @dai-shi ;

Thanks for your kind answer 😊

I'm actually giving a shot with raw jotai-tanstack-query and the vanilla Zodios API client nested inside; that will give me great answers on usage.

As for contribution / maintain; I believe this would be so close to jotai-tanstack-query that it could rely on it, and possibly be maintained next to it. Is the creator of jotai-tanstack-query still maintainer ? We could ask him if interested.

**Reply by dai-shi** [reply]
You may already know, but Jotai core is not tied to React. I think standalone jotai-zodios makes more senses, because the client api seems much smaller and cleaner to deal with.

jotai-tanstack-query is looking for a maintainer and it has some known limitations at the moment. https://github.com/jotaijs/jotai-tanstack-query/issues/42

**Reply by cyrilchapon** [reply]
@dai-shi thanks for the background. I was indeed talking about jotai-tanstack-query (and not jotai-core). I was hoping "the creator" (you ?) of jotai-tanstack-query could afford maintaining a light "layer" over it – precisely, someone who had the react-query knowledge you're asking for actually.

I'm actually not that profile unfortunately; and understood that this would first require the overhaul on jotai + react-query you're begging for.

I can surely help – because I like both sides of the project; but I lack knowledge on both sides so not able to take the lead by any means unfortunately =/

I might take a look to jotai-urql though, to appreciate the complexity and maybe come back with recommendations or a PoC

**Reply by dai-shi** [reply]
You can definitely learn from jotai-tanstack-query and jotai-urql,
but the fact that zodios uses react-query for react binding doesn't mean the jotai binding should be done like jotai-tanstack-query. It sounds like a wrong abstraction.
I do feel it should be close to jotai-trpc. The code might be very different but the abstraction should be.

**Reply by cyrilchapon** [reply]
Gotcha. I trust your experience with Jotai and will definitely give a look to jota-trpc.
Thanks again
